effort_controllers/src/joint_group_position_controller.cpp

```diff
@@ -0,0 +1,235 @@
+ *
+ * \section ROS interface
+ *
+ * \param type Must be "JointGroupEffortController".
+ * \param joints List of names of the joints to control.
+ *
+ * Subscribes to:
+ * - \b command (std_msgs::Float64MultiArray) : The joint efforts to apply
+ */
+  JointGroupPositionController::JointGroupPositionController() {}
+  JointGroupPositionController::~JointGroupPositionController() {sub_command_.shutdown();}
+
+  bool JointGroupPositionController::init(hardware_interface::EffortJointInterface* hw, ros::NodeHandle &n)
+  {
+    // List of controlled joints
+    std::string param_name = "joints";
+    if(!n.getParam(param_name, joint_names_))
+    {
+      ROS_ERROR_STREAM("Failed to getParam '" << param_name << "' (namespace: " << n.getNamespace() << ").");
+      return false;
+    }
+    n_joints_ = joint_names_.size();
+
+    if(n_joints_ == 0){
+      ROS_ERROR_STREAM("List of joint names is empty.");
+      return false;
+    }
+
+    // Get URDF
+    urdf::Model urdf;
+    if (!urdf.initParam("robot_description"))
+    {
+      ROS_ERROR("Failed to parse urdf file");
+      return false;
+    }
+
+    pid_controllers_.resize(n_joints_);
+
+    for(unsigned int i=0; i<n_joints_; i++)
+    {
+      try
+      {
+        joints_.push_back(hw->getHandle(joint_names_[i]));  
+      }
+      catch (const hardware_interface::HardwareInterfaceException& e)
+      {
+        ROS_ERROR_STREAM("Exception thrown: " << e.what());
+        return false;
+      }
+
+      urdf::JointConstSharedPtr joint_urdf = urdf.getJoint(joint_names_[i]);
+      if (!joint_urdf)
+      {
+        ROS_ERROR("Could not find joint '%s' in urdf", joint_names_[i].c_str());
+        return false;
+      }
+      joint_urdfs_.push_back(joint_urdf);
+
+      // Load PID Controller using gains set on parameter server
+      if (!pid_controllers_[i].init(ros::NodeHandle(n, joint_names_[i] + "/pid")))
+      {
+        ROS_ERROR_STREAM("Failed to load PID parameters from " << joint_names_[i] + "/pid");
+        return false;
+      }
+    }
+
+    commands_buffer_.writeFromNonRT(std::vector<double>(n_joints_, 0.0));
+
+    sub_command_ = n.subscribe<std_msgs::Float64MultiArray>("command", 1, &JointGroupPositionController::commandCB, this);
+    return true;
+  }
+
+  void JointGroupPositionController::update(const ros::Time& time, const ros::Duration& period)
+  {
+    std::vector<double> & commands = *commands_buffer_.readFromRT();
+    for(unsigned int i=0; i<n_joints_; i++)
+    {
+        double command_position = commands[i];
+//        double command_velocity = commands[i].velocity_;
+//        bool has_velocity_ =  commands[i].has_velocity_;
+
+        double error; //, vel_error;
+        double commanded_effort;
+
+        double current_position = joints_[i].getPosition();
+
+//TODO not sure if needed
+        // Make sure joint is within limits if applicable
+        //enforceJointLimits(command_position);
```

Comment:
position limites should get enforces the same way as in `JointPositionController`.